src/test/java/com/jwplayer/southpaw/MockState.java

```diff
@@ -38,13 +38,14 @@ public void configure(Map<String, Object> config) {
 
     @Override
     public void open() {
-        dataBatches = new HashMap<>();
         super.open();
     }
 
     @Override
     public void createKeySpace(String keySpace) {
-        dataBatches.put(new ByteArray(keySpace), new HashMap<>());
+        if(!dataBatches.containsKey(new ByteArray(keySpace))) {
+            dataBatches.put(new ByteArray(keySpace), new HashMap<>());
+        }
```

Comment:
```suggestion
        dataBatches.putIfAbsent(new ByteArray(keySpace), new HashMap<>());
```
I believe this can be simplified to just